packages/edit-site/src/components/navigation-sidebar/navigation-panel/search-results.js

```diff
@@ -7,25 +7,24 @@ import {
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
+/**
+ * Internal dependencies
+ */
+import { getTemplateInfo } from '../../../utils';
+import { normalizedSearch } from './utils';
+
 export default function SearchResults( {
 	items,
 	isDebouncing,
-	search,
-	children,
+	searchQuery,
+	renderItem,
 } ) {
 	const loading = items === null || isDebouncing;
 
-	if ( search.length === 1 ) {
-		return (
-			<NavigationGroup title={ __( 'Search results' ) }>
-				<NavigationItem
-					title={ __(
-						'Type at least 2 characters to start searching…'
-					) }
-				/>
-			</NavigationGroup>
-		);
-	}
+	const itemsFiltered = items.filter( ( item ) => {
+		const { title } = getTemplateInfo( item );
+		return normalizedSearch( title, searchQuery );
+	} );
```

Comment:
@Copons I'd love to use Controlled Search here but then I can't show "No results found" and "Loading" text. Any idea?

Reply:
Actually, I'm fine with this implementation but we should expose `normalizedSearch` or move it to an upper package so it's available for all the other packages. Rather than copy-pasting it everywhere :D

Reply:
The (un-)controlled search only works in a single `NavigationMenu`, making it not very viable for this use case.

About `normalizedSearch`: back when I added that I spent a while trying to figure out the best place to put it, in order to make it available across all packages. I haven't found anywhere that seemed appropriate, and then just settled on using an interal utility.
I'd be very happy to externalize it, but I honestly wouldn't know where. 😕